site/src/pages/tutorials/thrift/blog/implement-create.mdx

```diff
@@ -0,0 +1,160 @@
+    ...
+    blogPosts.put(id, blogPost);
+    final BlogPost stored = blogPost;
+    resultHandler.onComplete(stored);
+  }
+  ```
+
+## 2. Implement client-side
+
+In the `BlogClient` class, add a method to send a request to create a blog post.
+
+```java filename=BlogClient.java
+import example.armeria.blog.thrift.BlogPost;
+import example.armeria.blog.thrift.CreateBlogPostRequest;
+...
+BlogPost createBlogPost(String title, String content) throws TException {
+  final CreateBlogPostRequest request =
+          new CreateBlogPostRequest().setTitle(title)
+                                     .setContent(content);
+  return blogService.createBlogPost(request);
+}
+```
+
+## 3. Set up for testing
+
+Now, let's get started with the test code so that we can verify what we implement along the way.
+Note the code is minimal to keep things simple.
+
+1. Create a file, `BlogServiceTest.java`, and declare the test class.
+  You can see the full version of the file [here](https://github.com/line/armeria-examples/blob/main/tutorials/thrift/src/test/java/example/armeria/server/blog/thrift/BlogServiceTest.java).
+  ```java filename=BlogServiceTest.java
+  package example.armeria.server.blog.thrift;
+
+  class BlogServiceTest {}
+  ```
+2. Register a <type://ServerExtension> to set up a server instance for testing.
+  Note that the service instance is added to the configuration.
+  ```java filename=BlogServiceTest.java
+  import org.junit.jupiter.api.extension.RegisterExtension;
+
+  import com.linecorp.armeria.server.ServerBuilder;
+  import com.linecorp.armeria.server.thrift.THttpService;
+  import com.linecorp.armeria.testing.junit5.server.ServerExtension;
+
+  class BlogServiceTest {
+
+    @RegisterExtension
+    static final ServerExtension server = new ServerExtension() {
+      @Override
+      protected void configure(ServerBuilder sb) throws Exception {
+        sb.service("/thrift", THttpService.builder()
+                                          .addService(new BlogServiceImpl()) // Add the service to the configuration
+                                          .build());
+      }
+    };
+  }
+  ```
+
+## 4. Test creating a blog post
+
+Let's test if we can create a blog post.
+
+1. In the `BlogServiceTest` class, add a test method as follows.
+  ```java filename=BlogServiceTest.java
+  import static org.assertj.core.api.Assertions.assertThat;
+
+  import org.apache.thrift.TException;
+  import org.junit.jupiter.api.Order;
+  import org.junit.jupiter.api.Test;
+
+  import example.armeria.blog.thrift.BlogPost;
+
+  ...
+  @Test
+  void createBlogPost() throws TException {
+    final BlogClient client = new BlogClient(server.httpUri(), "/thrift");
+    final BlogPost response = client.createBlogPost("My first blog", "Hello Armeria!");
+    assertThat(response.getId()).isGreaterThanOrEqualTo(0);
+    assertThat(response.getTitle()).isEqualTo("My first blog");
+    assertThat(response.getContent()).isEqualTo("Hello Armeria!");
```

Comment:
I think assertions are good here because it's a test anyway. I agree that it can be useful to print the response as well. How about just doing both? 😄

Reply:
Adding both sounds good to me 🙂